### PR TITLE
feat(scripts): add script to install risingwave

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Add the initial script. Later once https://github.com/risingwavelabs/risingwave/issues/14898 is done, we can add other architectures and os as well.

The script will setup `meta_store` and `state_store` directories as well.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: scripts/install/install-risingwave.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  BASE_ARCHIVE_NAME="risingwave-${VERSION}-x86_64-unknown-linux-all-in-one"
+else
+  echo
+  echo "Unsupported OS or Architecture: ${OS}-${ARCH}"
+  echo
+  echo "Supported OSes: Linux"
+  echo "Supported architectures: x86_64"
+  echo
+  echo "Please open an issue at <https://github.com/risingwavelabs/risingwave/issues/new/choose>,"
+  echo "if you would like to request support for your OS or architecture."
```
Comment (kwannoel) [inline scripts/install/install-risingwave.sh:19]: Request arch support on demand

File: scripts/install/install-risingwave.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/sh -e
+
+OS=$(uname -s)
+ARCH=$(uname -m)
+VERSION=$(curl -s https://api.github.com/repos/risingwavelabs/risingwave/releases/latest | jq '.tag_name' -r)
```
Comment (fuyufjh) [inline scripts/install/install-risingwave.sh:5]: I am afraid that `jq` is not always available on user's env. Any alternatives? (how about https://gist.github.com/steinwaywhw/a4cd19cda655b8249d908261a62687f8)

Comment (kwannoel) [reply scripts/install/install-risingwave.sh:5]: True, I will just grep it.

**Comment by kwannoel** [comment]
- [ ] Change `jq` to grep
- [ ] Install for darwin via `brew`.
- [ ] Setup `.risingwave` in `$HOME`.

**Review by lmatz — APPROVED** [review]

File: scripts/install/install-risingwave.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  ARCHIVE_NAME="${BASE_ARCHIVE_NAME}.tar.gz"
+  URL="${BASE_URL}/${VERSION}/${ARCHIVE_NAME}"
+  USE_BREW=0
+elif [[ "${OS}" == "Darwin" ]] && [[ "${ARCH}" == "x86_64" || "${ARCH}" == "amd64" || "${ARCH}" == "aarch64" || "${ARCH}" == "arm64" ]]; then
+  USE_BREW=1
+else
+  echo
+  echo "Unsupported OS or Architecture: ${OS}-${ARCH}"
+  echo
+  echo "Supported OSes: Linux"
```
Comment (BugenZhao) [inline scripts/install/install-risingwave.sh:20]:
```suggestion
  echo "Supported OSes: Linux, macOS"
```
?

**Comment by kwannoel** [comment]
Will wait for amd release binary as well, then update the script before merging.

**Comment by kwannoel** [comment]
Can update the amd binary later, anyway we can always point to the script in latest main.

**Pull request merged**